R/hand_tiles.R

```diff
@@ -0,0 +1,413 @@
+            names(df) <- "hand"
+        }
+
+        locations <- sf::st_as_sf(
+            x = cbind(sf::st_coordinates(locations[, 1:2]), df),
+            coords = c("X", "Y"),
+            wkt = prj$wkt,
+        )
+    } else if (attributes(class(locations)) %in% c("raster")) {
+        if (
+            (is.null(loc_wkt) | nchar(loc_wkt) == 0 | is.na(loc_wkt)) &
+            is.null(prj)
+        ) {
+            stop("Please supply a valid WKT string.")
+        }
+
+        if (is.null(loc_wkt) | nchar(loc_wkt) == 0 | is.na(loc_wkt)) {
+            if (sum(!is.na(raster::getValues(locations))) == 0) {
+                stop("No distinct points, all values NA.")
+            } else {
+                locations <-
+                    raster::rasterToPoints(locations) %>%
+                    sf::st_as_sf(coords = c("x", "y"), wkt = prj$wkt)
+            }
+        }
+    }
+
+    locations
+}
+
+#' @title Project bounding box and Expand if needed
+#' @source Attribution: The `elevatr` R package.
+#'         \url{https://github.com/jhollist/elevatr}
+#' @keywords internal
+proj_expand <- function(locations, prj, expand) {
+
+    lll <- grepl("GEOGCRS", prj) |
+           grepl("GEODCRS", prj) |
+           grepl("GEODETICCRS", prj) |
+           grepl("GEOGRAPHICCRS", prj)
+
+    if (any(sf::st_bbox(locations)[c(2, 4)] == 0) & lll & is.null(expand)) {
+        # Edge case for lat exactly at the equator - was returning NA
+        expand <- 0.01
+    } else if (nrow(locations) == 1 & lll & is.null(expand)) {
+        # Edge case for single point and lat long
+        expand <- 0.01
+    } else if (nrow(locations) == 1 & is.null(expand)) {
+        # Edge case for single point and projected
+        expand <- 1
+    }
+
+    bbx <- sf::st_as_sfc(sf::st_bbox(locations))
+    if (!is.null(expand)) {
+        bbx <- sf::st_buffer(sf::st_as_sf(bbx), expand)
+    }
+
+    sf::st_bbox(bbx)
+}
+
+#' @title Clip the HAND raster
+#' @source Attribution: The `elevatr` R package.
+#'         \url{https://github.com/jhollist/elevatr}
+#' @keywords internal
+clip_it <- function(rast, loc, expand, clip) {
+    loc_wm <- sf::st_transform(loc, raster::crs(rast))
+    if (clip == "locations" & !grepl("Points", class(loc_wm))) {
+        hand <- raster::mask(raster::crop(rast, loc_wm), loc_wm)
+    } else if (clip == "bbox" | grepl("Points", class(loc_wm))) {
+        bbx <- proj_expand(loc_wm, as.character(raster::crs(rast)), expand)
+        bbx_sf <- sf::st_transform(
+            sf::st_as_sf(sf::st_as_sfc(bbx)),
+            raster::crs(rast)
+        )
+        hand <- raster::mask(raster::crop(rast, bbx_sf), bbx_sf)
+    }
+
+    hand
+}
+
+#' @title Estimate download size of DEMs
+#' @source Attribution: The `elevatr` R package.
+#'         \url{https://github.com/jhollist/elevatr}
+#' @param locations the locations
+#' @param z zoom level if source is aws
+#' @keywords internal
+estimate_raster_size <- function(locations, z = NULL) {
```

Comment:
We only have 1 res so is this accurate?

Reply:
I believe our zoom level is roughly between `z = 13` and `z = 14` (based on [this](https://wiki.openstreetmap.org/wiki/Zoom_levels)).

I will set this manually; although, the way this is coded, the zoom level doesn't seem to matter... (unless I'm misunderstanding it, since the `res` will be the same for any `z`)

Alternatively, I think each tile is exactly 256kb, so we could just pass the number of tiles and multiply that by 256 to get the total size in kb. *Does that seem better?*